src/main/java/synapseawsconsolelogin/Auth.java

```diff
@@ -405,11 +411,14 @@ static int synapseExceptionStatus(SynapseServerException e) {
 		// and we don't expect others. For the rest we'll just return 500.
 		return 500;
 	}
-
+	
 	@Override
 	public void doGet(HttpServletRequest req, HttpServletResponse resp)
 			throws IOException {
 		try {
+			if (req.isSecure()) {
```

Comment:
i'm not sure this check is necessary since it seems like browsers will ignore the header if http is used..  https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Strict-Transport-Security#description

Reply:
The check might not be necessary, however (from the page you link):

> The Strict-Transport-Security header is ignored by the browser when your site has only been accessed using HTTP. Once your site is accessed over HTTPS with no certificate errors, the browser knows your site is HTTPS capable and will honor the Strict-Transport-Security header. Browsers do this as attackers may intercept HTTP connections to the site and inject or remove the header.

So there's no point in adding the header for `HTTP` requests.